### ISSUE TITLE
Update test: make sure email gets deleted from returned values

### ISSUE THREAD

**Issue by acabunoc** [mozillascience/PaperBadger#66]
To edit: https://github.com/mozillascience/PaperBadger/blob/master/test/helpers.js#L17
Mentor: @acabunoc

https://github.com/mozillascience/PaperBadger/commit/3c80356278a15886687d8a82e7cfd2a9febd7a5b

From #58

**Comment by acabunoc** [comment]
Current test: https://github.com/mozillascience/PaperBadger/blob/master/test/helpers.js#L17

This passes even if email gets returned in api.

**Comment by guahanweb** [comment]
Just ran this locally, and it seems to work just fine when I return email in the payload:
![image](https://cloud.githubusercontent.com/assets/155616/9282223/b4d51a30-427f-11e5-9046-41e2d6e59c0d.png)

What is the issue?

**Comment by acabunoc** [comment]
Just re-read the issue - sorry it's so cryptic!

Some background: https://github.com/mozillascience/PaperBadger/pull/58

In our code, `entry` is normally an object returned from `badgekit-api-client` (not just a hash like in the test). When we run `delete entry.email;` it didn't actually delete that key for some reason. We need to update the test to reflect what's actually going on in our code.

**Comment by guahanweb** [comment]
So... it feels like this test should use a mock response for the endpoint that builds the actual email object. That way, we're testing against the actual object. Is that feasible here? Where is the code that defines the client object that gets returned?

**Comment by guahanweb** [comment]
OK... after a bit more investigation, I'm proposing a bit of a restructure for the helper that does the modification. I diagnosed the response, and it looks like the helpers are suffering from references not being updated.

I restructured the helpers a bit to actually clone the response, setting only the appropriate variables. This changes the response slightly, but I updated all the tests to reflect it, too. Let me know if there are any concerns about this:

https://github.com/mozillascience/PaperBadger/pull/88
